synapse/app/generic_worker.py

```diff
@@ -458,21 +458,22 @@ def start(config_options: List[str]) -> None:
         config.appservice.notify_appservices = False
 
     if config.worker.worker_app == "synapse.app.user_dir":
-        if config.server.update_user_directory:
+        from synapse.config.server import ANY_USER_DIRECTORY_WORKER
```

Comment:
I think there's a preference for having imports at the top of the file where possible.

Reply:
This is to avoid cyclic dependencies, but i think it could be okay enough, let me check.